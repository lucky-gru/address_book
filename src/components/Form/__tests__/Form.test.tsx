import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { Form, FormGroup, Error } from '../index'

test('Form render', () => {
  const tree = renderer.create(<Form />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('margin-top', '24px')
})

test('FormGroup render', () => {
  const tree = renderer.create(<FormGroup />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('margin-bottom', '1rem')
})

test('Error render', () => {
  const tree = renderer.create(<Error>really error</Error>).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('color', '#ff4c50')
})
