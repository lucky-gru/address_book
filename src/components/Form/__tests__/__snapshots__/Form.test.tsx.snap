// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error render 1`] = `
.c0 {
  margin: 15px 0 0 10px;
  display: inline-block;
  padding: 5px;
  border: 1px solid #ff4c50;
  color: #ff4c50;
  font-size: 1rem;
}

<span
  className="c0"
>
  really error
</span>
`;

exports[`Form render 1`] = `
.c0 {
  margin-top: 24px;
}

<div
  className="c0"
/>
`;

exports[`FormGroup render 1`] = `
.c0 {
  margin-bottom: 1rem;
  width: 100%;
}

.c0 label {
  font-size: 1rem;
  font-family: 'Gotham';
  font-weight: 500;
  display: block;
  margin: 0 0 5px 15px;
}

.c0 > input[type='text'] {
  width: 100%;
  height: 42px;
  line-height: 42px;
  padding: 0 12px;
  border-radius: 21px;
  display: block;
  background: #fff;
  border: 1px solid #c8c8c8;
  font-size: 1rem;
  font-family: 'Gotham';
  font-weight: 500;
  color: #343a40;
}

.c0 > span {
  margin: 5px 0 0 10px;
  display: inline-block;
  padding: 5px;
  border: 1px solid #ff4c50;
  color: #ff4c50;
  font-size: 1rem;
}

<div
  className="c0"
/>
`;
